src/main/java/sirius/kernel/xml/XMLCall.java

```diff
@@ -128,16 +147,46 @@ public XMLStructuredOutput getOutput() throws IOException {
         return new XMLStructuredOutput(outcall.postFromOutput());
     }
 
+    private InputStream getInputStream() throws IOException {
+        if (logger != null && logger.isFINE()) {
+            // log the request, even when parsing fails
+            try (InputStream body = outcall.getResponse().body()) {
+                String text = new String(body.readAllBytes(), StandardCharsets.UTF_8);
+                boolean isPost = outcall.isPostRequest();
+                logger.FINE(Formatter.create("""
+                                                     ---------- call ----------
+                                                     ${httpMethod} ${url} [
+                                                                                  
+                                                     ${callBody}]
+                                                     ---------- response ----------
+                                                     HTTP-Response-Code: ${responseCode}
+                                                                                  
+                                                     ${response}
+                                                     ---------- end ----------
+                                                     """)
+                                     .set("httpMethod", isPost ? "POST" : "GET")
+                                     .set("url", outcall.getRequest().uri())
+                                     .set("callBody", isPost ? getOutput() : null)
+                                     .set("responseCode", getOutcall().getResponseCode())
+                                     .set("response", text)
+                                     .smartFormat());
+                return new ByteArrayInputStream(text.getBytes(StandardCharsets.UTF_8));
+            }
+        }
+        return outcall.getResponse().body();
+    }
+
     /**
      * Provides access to the XML answer of the call.
      *
      * @return the XML result of the call
      * @throws IOException in case of an IO error while receiving the result
      */
     public XMLStructuredInput getInput() throws IOException {
+        InputStream body = getInputStream();
         String contentType = outcall.getHeaderField("content-type");
         if (!outcall.isErroneous() || (contentType != null && contentType.toLowerCase().contains("xml"))) {
-            return new XMLStructuredInput(outcall.getResponse().body(), namespaceContext);
+            return new XMLStructuredInput(body, namespaceContext);
```

Comment:
remove the extra variable and call in place (?)

Reply:
I'd like to leave it here so failures can be logged as well.